src/main/kotlin/build/buf/gradle/BuildConfiguration.kt

```diff
@@ -52,13 +54,14 @@ internal fun Project.configureImagePublication(artifactDetails: ArtifactDetails)
 
             artifact(bufBuildPublicationFile) {
                 builtBy(tasks.named(BUF_BUILD_TASK_NAME))
+                extension = specifiedExtension
```

Comment:
Do you know what purpose this property serves and why it's necessary now?

Reply:
Without telling the extension to maven-publish plugin, the published image file extension would be incorrect if you specify an image extension with a compression as an option. For example, if you choose `json.gz` as an option and expect to get `image.json.gz`, you would get `image.gz`.

Reply:
Makes sense 👍